**Heart Disease**

#**Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that enable us to be able to

In [19]:
import pandas as pd
import numpy as np
import plotly.express as px
import matplotlib.pyplot as plt


#**Context**

Coronary heart disease (CHD) involves the reduction of blood flow to the heart muscle due to the build-up of plaque in the arteries of the heart. It is the most common form of cardiovascular disease. Currently, invasive coronary angiography represents the gold standard for establishing the presence, location, and severity of CAD, however, this diagnostic method is costly and associated with morbidity and mortality in CAD patients. Therefore, it would be beneficial to develop a non-invasive alternative to replace the current gold standard.

Other less invasive diagnostics methods have been proposed in the scientific literature including exercise electrocardiogram, thallium scintigraphy, and fluoroscopy of coronary calcification. However, the diagnostic accuracy of these tests only ranges between 35%-75%. Therefore, it would be beneficial to develop a computer-aided diagnostic tool that could utilize the combined results of these non-invasive tests in conjunction with other patient attributes to boost the diagnostic power of these non-invasive methods with the aim of ultimately replacing the current invasive gold standard.

A number of 303 consecutive patients referred for coronary angiography at the Cleveland Clinic between May 1981 and September 1984 participated in the experiment. No patient had a history or electrocardiographic evidence of prior myocardial infarction or known valvular or cardiomyopathic diseases.


#**About the Dataset**

The dataset comprises 303 observations, 13 features, and 1 target attribute. The 13 features include the results of the aforementioned non-invasive diagnostic tests along with other relevant patient information. The target variable includes the result of the invasive coronary angiogram which represents the presence or absence of coronary artery disease in the patient. The 14 variables (13 features and 1 target attribute) are described below.

| **Item**    | **Description**                                                                                   |
|-------------|---------------------------------------------------------------------------------------------------|
| AGE         | Displays the age of the individual.                                                               |
| SEX         | Displays the gender of the individual using the following format: 1 = male, 0 = female.          |
| CP          | Displays the type of chest pain experienced by the individual:                                    |
|             | - 0 = typical angina                                                                             |
|             | - 1 = atypical angina                                                                            |
|             | - 2 = non-anginal pain                                                                           |
|             | - 3 = asymptotic                                                                                 |
| TRESTBPS    | Displays the resting blood pressure value of an individual in mmHg (unit).                        |
| CHOL        | Displays the serum cholesterol in mg/dl (unit).                                                  |
| FBS         | Compares the fasting blood sugar value of an individual with 120mg/dl:                             |
|             | - 1: fasting blood sugar >120mg/dl                                                               |
|             | - 0: fasting blood sugar ≤ 120mg/dl                                                              |
| RESTECG     | Displays resting electrocardiographic results:                                                   |
|             | - 0 = normal                                                                                     |
|             | - 1 = having ST-T wave abnormality                                                               |
|             | - 2 = left ventricular hypertrophy                                                               |
| THALACH     | Displays the max heart rate achieved by an individual.                                           |
| EXANG       | Exercise-induced angina:                                                                         |
|             | - 1 = yes                                                                                        |
|             | - 0 = no                                                                                         |
| OLDPEAK     | ST depression induced by exercise relative to rest. Displays the value (integer or float).       |
| SLOPE       | Peak exercise ST segment:                                                                        |
|             | - 1 = upsloping                                                                                  |
|             | - 2 = flat                                                                                       |
|             | - 3 = downsloping                                                                                |
| CA          | Number of major vessels (0-3) colored by fluoroscopy. Displays the value (integer or float).      |
| THAL        | Displays the thalassemia:                                                                       |
|             | - 1 = normal                                                                                     |
|             | - 2 = fixed defect                                                                              |
|             | - 3 = reversible defect                                                                         |
| TARGET      | Displays whether the individual is suffering from heart disease or not:                           |
|             | - 0 = absence                                                                                   |
|             | - 1 = present                                                                                   |





**Data**

In [20]:
#url="https://archive.ics.uci.edu/static/public/45/data.csv"
#df=pd.read_csv(url)
df=pd.read_csv("Heart_Disease.csv")
df

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
0        0   0     1       1  
1        0   0     2       1  
2        2   0     2       1  
3        2   0     2       1  
4        2   0     2       1  
..     ...  ..   ...     ...  
298      1   0     3       0  
299      1   0     3       0  
300      1   2     3       0  
301      1   1     3       0  
302      1   1     2       0  

[303 rows x 14 columns]

Next, we can display the data by typing the name of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [21]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
0        0   0     1       1  
1        0   0     2       1  
2        2   0     2       1  
3        2   0     2       1  
4        2   0     2       1  
..     ...  ..   ...     ...  
298      1   0     3       0  
299      1   0     3       0  
300      1   2     3       0  
301      1   1     3       0  
302      1   1     2       0  

[303 rows x 14 columns]

#**ASSIGNMENT 1 - Descriptive Statistics: Graphical and Numerical Summary**

**INSTRUCTIONS**

Use Python to analyze the data set and complete each of the following. As appropriate, copy the SPSS output and paste it in the correct part below. For problems that require a written response, type the answer below.

##**QUESTION 1**

Determine whether the three variables below are qualitative or quantitative. If they are quantitative, specify whether they are continuous or discrete.

| **Variable**              | **Type**          | **Description**                                      |
|---------------------------|-------------------|------------------------------------------------------|
| **Age**                   | Quantitative      | Continuous age of the individual.                    |
| **Sex**                   | Qualitative       | Gender of the individual (1 = male, 0 = female).     |
| **Chest Pain Type (CP)**  | Qualitative       | Type of chest pain experienced by the individual:    |
|                           |                   | - 0 = typical angina                                 |
|                           |                   | - 1 = atypical angina                                |
|                           |                   | - 2 = non-anginal pain                               |
|                           |                   | - 3 = asymptotic                                     |
| **Resting BP (TRESTBPS)** | Quantitative      | Resting blood pressure value in mmHg (unit).         |


##**QUESTION 2**

Construct a frequency table, relative frequency table, and relative frequency bar chart to describe the distribution of chest pain type. State any fact that jumps out to you.

In [ ]:
#Frequency table
freq_table = pd.value_counts(df['cp'])
freq_table

In [ ]:
#Relative frequency table
relative_freq_table=freq_table/len(df)
relative_freq_table

In [23]:
freq_table = pd.value_counts(df['cp']).sort_index()
relative_freq=((freq_table/len(df))*100).round(1)
valid_percent=relative_freq.copy()
cumulative_percent=relative_freq.cumsum().round(2)

final_table = pd.DataFrame({
    'Frequency': freq_table,
    'Percent': relative_freq,
    'Valid Percent': valid_percent,
    'Cumulative Percent': cumulative_percent})

final_table.loc['Total']=final_table.sum()
print(final_table)

       Frequency  Percent  Valid Percent  Cumulative Percent
0          143.0     47.2           47.2                47.2
1           50.0     16.5           16.5                63.7
2           87.0     28.7           28.7                92.4
3           23.0      7.6            7.6               100.0
Total      303.0    100.0          100.0               303.3


In [9]:
frequency_table = df['cp'].value_counts(normalize=True).sort_index()

frequency_table.index = frequency_table.index.astype(str)
frequency_table_percent = frequency_table * 100

fig = px.bar(x=frequency_table.index, y=frequency_table_percent.values,
             title='Relative Frequency Distribution Bar Chart',
             labels={'x':'Chest Pain Type', 'y':'Percent'})
fig.update_xaxes(type='category', categoryorder='array', categoryarray=[0, 1, 2, 3])  # Order x-axis
fig.show()

47% of patients have atypical angina; 16.5% of patients have atypical angina; 28.7% of patients have non anginal pain and 7.6% have asymptotic pain.

## **QUESTIONS 3-6**

For questions 3-6: Find your variable based on your last name and use that variable when answering questions #3 to #6.  

| **Last Name Variable** | **Description**                                      |
|------------------------|------------------------------------------------------|
| A-F                    | Resting blood pressure                    |
| G-M                    | Serum cholesterol                              |
| N-S                    | Maximum heart rate achieved                |
| T-Z                    | ST depression induced by exercise relative to rest  |



###**QUESTION 3**

Construct a histogram for your variable. Use Number of Intervals = 15.

In [ ]:
fig = px.histogram(x=df['trestbps'],nbins = 15)
fig.show()

In [ ]:
fig = px.histogram(x=df['chol'],nbins = 15)
fig.show()

In [ ]:
fig = px.histogram(x=df['thalach'],nbins = 15)
fig.show()

In [ ]:
fig = px.histogram(x=df['oldpeak'],nbins = 15)
fig.show()

###**QUESTION 4**

Construct a boxplot for your variable.  

In [ ]:
px.box(y=df['trestbps'])

In [ ]:
px.box(y=df['chol'])

In [ ]:
px.box(y=df['thalach'])

In [ ]:
px.box(y=df['oldpeak'])

###**QUESTION 5**

Calculate the following summary statistics for your variable: minimum, maximum, mean, median, standard deviation, Q1, and Q3. Paste the output below.

In [ ]:
descriptive_stats = df[['cp','trestbps','chol','thalach','oldpeak']].describe().round(3)
descriptive_stats


cp  trestbps     chol  thalach  oldpeak
count  303.000   303.000  303.000  303.000  303.000
mean     0.967   131.624  246.264  149.647    1.040
std      1.032    17.538   51.831   22.905    1.161
min      0.000    94.000  126.000   71.000    0.000
25%      0.000   120.000  211.000  133.500    0.000
50%      1.000   130.000  240.000  153.000    0.800
75%      2.000   140.000  274.500  166.000    1.600
max      3.000   200.000  564.000  202.000    6.200

###**QUESTION 6**

Use information from questions #3, #4, and #5 to describe your variable in terms of shape, center, spread, and outliers. Interpret your findings.

The distribution of blood pressure is skewed right. The median is 130 mmHg and IQR is 20 mmHg. There are 7 outliers.

The distribution of serum cholesterol is skewed right. The median is 240 mg/dl and IQR is 64 mg/dl. There are 5 outliers.

The distribution of the maximum heart rate achieved is skewed left. The median is 153 mmHg and IQR is 33 mmHg. There is 1 outlier.

The distribution of ST depression induced by exercise relative to rest is skewed right. The median is 0.8 and IQR is 1.6. There are 4 outliers.

##**QUESTION 7**

Calculate and state the mean age, mean resting blood pressure, and mean cholesterol for those with heart disease, and for those without heart disease. Compare the results.

In [ ]:
heart_disease_means = df[df['target'] == 1][['age', 'trestbps', 'chol']].mean().round(2)
no_heart_disease_means = df[df['target'] == 0][['age', 'trestbps', 'chol']].mean().round(2)
total_means = df[['age', 'trestbps', 'chol']].mean().round(2)

sec_stat = pd.DataFrame({'age': [no_heart_disease_means['age'], heart_disease_means['age'], total_means['age']],
                         'trestbps': [no_heart_disease_means['trestbps'], heart_disease_means['trestbps'], total_means['trestbps']],
                         'chol': [no_heart_disease_means['chol'], heart_disease_means['chol'], total_means['chol']]},
                        index=['0', '1', 'Total'])

print(sec_stat)

         age  trestbps    chol
0      56.60    134.40  251.09
1      52.50    129.30  242.23
Total  54.37    131.62  246.26


Patients without heart disease tended to be older, with higher blood pressure and cholesterol levels.

##**QUESTION 8**

Generate a paragraph of at least 100 words to address one of the following questions:


### **QUESTION 8a**

Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

In upper division STAT courses, I frequently have to use descriptive statistics and visualization to display datasets. Knowing how to generate these graphs and tables can make it easier to grasp the dataset that I am working with.  It is much easier to determine the distribution of data once it is organized in a table or visualized in a graph. I am currently taking STAT473 (Statistical Machine Learning) and STAT495, (Statistical Modeling in Epidemiology), and I find that visualizing the dataset and understanding its patterns makes it easier to determine if your calculations are correct. Oftentimes, it can be difficult to check your answers if you do not understand what the statistical methods represent.

### **QUESTION 8b**

Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?

I want to pursue a Statistics related career in the future, so it is essential to utilize statistical methods in my work. Having a strong foundation in applying statistical methods (such as generating descriptive statistics and visualizing data) will allow me to efficiently find patterns in data and make more informed decisions. Understanding what statistical methods represent will grant me domain-specific knowledge and allow me to view the data in real-world context. Without a solid understanding of statistical methods, I wouldn't be able to pursue my career at all, as I would not be able to draw conclusions with datasets.